apis/launchers.md

```diff
@@ -63,46 +63,69 @@ As you can see, the method `set_progress` takes a `double` value type and is a r
 [...]
-Actions=ActionID;
+DBusActivatable=true
 ```
 
-Then below, add the action itself using the same unique ID:
+### Declaring actions
+
+You can use any action defined in the `app` namespace, i.e. registered with `GLib.Application`, as an entry point for your application. Implementing actions is covered in-depth in [the actions section](actions). They must also be declared in a new `Actions` line in your app's `.desktop` file. This line should contain a `;` separated list of action names:
 
 ```ini
-[Desktop Action ActionID]
-Name=The name of the action
-Icon=com.github.myteam.myapp.action-id
-Exec=com.github.myteam.myapp --action-id
+[Desktop Entry]
+Name=Hello Again
+[...]
+Actions=my-action;
 ```
 
-The `Icon` line is optional and should be an icon which represents the action that will be performed. The `Exec` line is required and should be your app's executable name and any command line argument required to trigger the action.
-
-{% hint style="info" %}
-The action name should not include your app's name, as it will always be displayed alongside your app. The action icon should also not be your app icon, as it may be shown in the menu for your app icon, or badged on top of the app icon.
-{% endhint %}
-
-Let's take a look at an example of an action that opens a new window of an app:
+Then use a dedicated group, named after the unique action name, to define the details of each action:
 
 ```ini
-[Desktop Entry]
-Name=App Name
-Exec=com.github.yourusername.yourrepositoryname
-...
-Actions=NewWindow;
-
-[Desktop Action NewWindow]
-Name=New Window
-Exec=com.github.yourusername.yourrepositoryname -n
+[Desktop Action my-action]
+Name=My Great Action
+Icon=com.github.myteam.myapp.my-action-icon
+Exec=com.github.myteam.myapp
 ```
 
-Note that just adding `-n` or any other argument will not automatically make your app open a new window; your app must handle and interpret command line arguments. The [GLib.Application API](https://valadoc.org/gio-2.0/GLib.Application.html) provides many examples and an extensive documentation on how to handle these arguments, particularly the [command\_line signal](https://valadoc.org/gio-2.0/GLib.Application.command\_line.html).
+{% hint style="warning" %}
+The action name used in `.desktop` file, both in Desktop Entry and later in Desktop Action groups, needs to match exactly the name used to register the action with `GLib.Applicatin` in the source code.
+{% endhint %}
+
+The `Icon` line is optional and should be an icon which represents the action that will be performed. The `Exec` line should be specified, but is used only for backwards compatibility in case your app ever runs in an environment without D-Bus activation support.
+
+{% hint style="info" %}
+The action name should not include your app's name, as it will always be displayed alongside your app. The action icon should also not be your app icon, as it may be shown in the menu for your app icon, or badged on top of the app icon.
+{% endhint %}
 
```

Comment:
Can we add a link in here to the code sample like we do in the Actions section?

Reply:
Yes! I [added the link](https://github.com/elementary/docs/pull/185/commits/20d69b1afe41af0745fb26d7f287360190b61f4a) to the bottom of the page. I also flattened this text here about action names and icons from info box into a standard paragraph. I was afraid that having every second paragraph be styled as info or warning would be distracting.